packages/components/src/internal/query/APIWrapper.ts

```diff
@@ -107,9 +109,19 @@ export interface QueryAPIWrapper {
     ) => Promise<void>;
     selectDistinctRows: (selectDistinctOptions: Query.SelectDistinctOptions) => Promise<Query.SelectDistinctResponse>;
     selectRows: (options: SelectRowsOptions) => Promise<SelectRowsResponse>;
+    setSelected: (
```

Comment:
nit: This is too many parameters for a method. I would convert this accept an "options" paradigm.